scripts/update-snapshot-id-in-launch-conf.py

```
@@ -0,0 +1,173 @@
+#!/usr/bin/env python
+import sys, os, logging, argparse, socket, textwrap, boto3
+
+__version__='0.1'
+try:
+  import coloredlogs
+  coloredlogs.install(
+    isatty = True,
+    show_name = False,
+    show_severity = False,
+    level = logging.NOTSET,
+    severity_to_style = { 'DEBUG': {'color': 'blue'}},
+  )
+except:
+  logging.basicConfig(
+    stream = sys.stdout,
+    format = '%(asctime)s ' + socket.gethostname() + ' %(levelname)-8s %(message)s',
+    datefmt = "%Y-%m-%d %H:%M:%S",
+    level = logging.NOTSET,
+  )
+log = logging.getLogger(__name__)
+
+def find_autoscaling_group(client, component, stack_prefix):
+  count_found = 0
+  groups = client.describe_auto_scaling_groups()['AutoScalingGroups']
+  for group in groups:
+    for tag in group['Tags']:
+      if (tag['Key'] == 'StackPrefix' and tag['Value'] == stack_prefix) or \
+          (tag['Key'] == 'Component' and tag['Value'] == component):
+        if count_found == 1:
+          log.debug('Autoscaling Group: %r', group)
+          return group
+        else:
+          count_found += 1
+
```

I might need to throw an exception if the device_name is not found...